packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx

```diff
@@ -809,40 +784,98 @@ export class ScatterPlotChart
-                        subtitle={timeLabel}
-                        dissolve={fading}
-                        notice={targetNotice}
-                    >
-                        <TooltipValueRange
-                            column={this.xColumn}
-                            values={values.map((v) => v.x)}
-                            notice={xNotice}
-                        />
-                        <TooltipValueRange
-                            column={this.yColumn}
-                            values={values.map((v) => v.y)}
-                            notice={yNotice}
-                        />
-                        <TooltipValueRange
-                            column={this.sizeColumn}
-                            values={compact(values.map((v) => v.size))}
-                        />
-                    </Tooltip>
-                )}
+                {this.tooltip}
             </g>
         )
     }
 
+    @computed get tooltip(): JSX.Element | null {
+        if (!this.tooltipState.target) return null
+
+        const {
+            xColumn,
+            yColumn,
+            tooltipState: { target, position, fading },
+        } = this
+        const points = target.series.points ?? []
+        const values = compact(uniq([first(points), last(points)]))
+
+        let { startTime, endTime } = this.manager
+        const { x: xStart, y: yStart } = first(values)?.time ?? {},
+            { x: xEnd, y: yEnd } = last(values)?.time ?? {}
+
+        let xValues = values.map((v) => v.x),
+            xNoticeNeeded =
+                (xStart !== undefined && xStart != startTime && xStart) ||
+                (xEnd !== undefined && xEnd != endTime && xEnd),
+            xNotice = xNoticeNeeded ? [xStart, xEnd] : []
+
+        let yValues = values.map((v) => v.y),
+            yNoticeNeeded =
+                (yStart !== undefined && yStart != startTime && yStart) ||
+                (yEnd !== undefined && yEnd != endTime && yEnd),
+            yNotice = yNoticeNeeded ? [yStart, yEnd] : []
+
+        // handle the special case where the same variable is used for both axes
+        // with a different year's value on each
+        if (xColumn.def.datasetId === yColumn.def.datasetId) {
+            const xTime = xColumn.def.targetTime ?? this.manager.endTime,
+                yTime = yColumn.def.targetTime ?? this.manager.endTime
+
+            if (xTime != yTime && isNumber(xTime) && isNumber(yTime)) {
+                startTime = min([xTime, yTime])
+                endTime = max([xTime, yTime])
+                xValues =
+                    xTime < yTime
+                        ? [values[0]?.x, values[0]?.y]
+                        : [values[0]?.y, values[0]?.x]
+                xNotice = yNotice = yValues = []
+                xNoticeNeeded = yNoticeNeeded = false
+            }
+        }
+
+        const { isRelativeMode } = this.manager,
+            timeRange = uniq(compact([startTime, endTime]))
```

Comment:
The above comment about `compact` actually applies here: If one of these times is `0` then we're gonna filter it out by accident.